### PR TITLE
Bufr Converter m x n Problem Bug Fix

### PR DESCRIPTION
**Pull request by rmclaren**
## Description
Fixes a bug where m x n data was not being interpretted properly when both m and n are > 1. Also has some minor fixes to run_satwnds stuff.

### Issue(s) addressed

#413

### CODE REVIEW

File: src/bufr/BufrParser/BufrCollectors/BufrAccumulator.cpp [hunk 1]
```diff
@@ -32,19 +32,21 @@ namespace Ingester
...
     }
 
-    IngesterArray BufrAccumulator::getData(Eigen::Index startCol, const Channels &channels)
+    IngesterArray BufrAccumulator::getData(Eigen::Index elementPos,
+                                           Eigen::Index numElementsPerSet,
+                                           const Channels& indicies)
     {
-        if (std::find_if(channels.begin(), channels.end(), [](const auto x){ return x < 1; }) \
-            != channels.end())
+        if (std::find_if(indicies.begin(), indicies.end(), [](const auto x){ return x < 1; }) \
```
Comment (CoryMartin-NOAA) [inline src/bufr/BufrParser/BufrCollectors/BufrAccumulator.cpp:39]: "indices" :-)

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -514,6 +517,14 @@ if(iodaconv_bufr_ENABLED)
                             gdas.t06z.adpsfc.tm00.nc ${IODA_CONV_COMP_TOL_ZERO}
                     DEPENDS bufr2ioda.x )
 
+  ecbuild_add_test( TARGET  test_iodaconv_bufr_m_x_n
```
Comment (CoryMartin-NOAA) [inline test/CMakeLists.txt:520]: is there a better name? this means almost nothing to me... Or perhaps describe what the test does in the YAML?

**Review by CoryMartin-NOAA — CHANGES_REQUESTED** [review]

File: test/CMakeLists.txt (hunk 2 above)
Comment (rmclaren) [reply test/CMakeLists.txt]: Fixed Spelling.

**Comment by rmclaren** [comment]
@CoryMartin-NOAA I renamed the unit test to **test_iodaconv_bufr_read_2_dim_blocks**.

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks good to me, thanks for addressing my comments!

**Comment by srherbener** [comment]
When I build and test on my iMac, the `test_iodaconv_bufr_read_2_dim_blocks` appears to be stuck or executing very slowly. The other tests seem okay.
```
sysadmins-iMac:build stephenh$ ctest -R test_iodaconv_bufr
Test project /Users/stephenh/projects/BUFR_CONV/ioda-bundle/build
    Start 478: test_iodaconv_bufr_coding_norms
1/9 Test #478: test_iodaconv_bufr_coding_norms ........   Passed    1.20 sec
    Start 505: test_iodaconv_bufr_bufrdescription
2/9 Test #505: test_iodaconv_bufr_bufrdescription .....   Passed    0.09 sec
    Start 506: test_iodaconv_bufr_mhs2ioda
3/9 Test #506: test_iodaconv_bufr_mhs2ioda ............   Passed    1.98 sec
    Start 507: test_iodaconv_bufr_hrs2ioda
4/9 Test #507: test_iodaconv_bufr_hrs2ioda ............   Passed    1.04 sec
    Start 508: test_iodaconv_bufr_filtering
5/9 Test #508: test_iodaconv_bufr_filtering ...........   Passed    0.52 sec
    Start 509: test_iodaconv_bufr_splitting
6/9 Test #509: test_iodaconv_bufr_splitting ...........   Passed    0.48 sec
    Start 510: test_iodaconv_bufr_filter_split
7/9 Test #510: test_iodaconv_bufr_filter_split ........   Passed    0.46 sec
    Start 511: test_iodaconv_bufr_adpsfc2ioda
8/9 Test #511: test_iodaconv_bufr_adpsfc2ioda .........   Passed    3.22 sec
    Start 512: test_iodaconv_bufr_read_2_dim_blocks
```

The test `test_iodaconv_bufr_read_2_dim_blocks` timed out after 1500 sec.

I'm guessing that there might be something wrong with my iMac. Are you seeing this test run in reasonable time on other systems?

Please note that the CI testing is currently not building the EMC converter and thus not running its tests. The work to get CI testing to include the EMC converter is in a draft PR right now (#416).

**Comment by CoryMartin-NOAA** [comment]
@srherbener let me clone/build on Hera this afternoon to confirm it works.

**Comment by CoryMartin-NOAA** [comment]
@srherbener @rmclaren yeah it's hanging for me too on Hera, good thing you checked. I was too hasty with my approval. Looks like it is a problem with the comparison script? As just running this: "/scratch1/NCEPDEV/stmp2/Cory.R.Martin/ioda/ioda-bundle/build/bin/bufr2ioda.x testinput/bufr_read_2_dim_blocks.yaml" finishes in a couple seconds

**Comment by rmclaren** [comment]
Thats weird, it works on my Ubuntu machine... I'll try it on Hera as well.

**Comment by CoryMartin-NOAA** [comment]
hanging on nccmp for some reason it looks like, how big is this file?

**Comment by CoryMartin-NOAA** [comment]
@rmclaren do we need 98000 locations for this? Not sure if that is the problem, but what about reducing that by a factor of 1000?

**Comment by rmclaren** [comment]
Not very big, about 870KB from the last time I ran it (for the netcdf file).

**Comment by CoryMartin-NOAA** [comment]
adding debug information to nccmp, it's taking a REALLY long time to get through the 12x97883 variable for some reason.

**Comment by rmclaren** [comment]
I'll try to run Rahuls data reduction function to reduce the number of location. The input bufr file has already been split so that only one type of subset is represented, so its already pretty small,but still has all the original locations.

**Comment by rmclaren** [comment]
Ok, I reduced the size of the bufr file to 10000 records (200 messages). Size wise this is comparable to the other bufr files we use in tests, so hopefully it will be less of an issue.

**Comment by rmclaren** [comment]
The smaller file does work for me on Hera as well, though it takes a little longer to run than it does on my local machine (0.8 seconds local 1.3 seconds HERA)

**Comment by CoryMartin-NOAA** [comment]
Yes the test works for me now too, although 5 seconds is still a really long time. But I guess it's probably okay.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by srherbener** [comment]
@rmclaren thanks for trimming the input bufr file. The test runs in 1.5 sec on my iMac now. I think this is good for now.

We are going to introduce this month a new way to manage test files (for unit testing only) using a ioda-data repo and git-lfs. We want to put only unit test appropriate files there. The goal of unit testing is to get high coverage of code with very short fast running tests.

I would say that this file with 10000 locations is still too big for unit testing. However, I'm okay merging now and addressing this later since your file isn't the only one that is perhaps "too big" for unit testing.

**Review by srherbener — APPROVED** [review]

**Pull request merged**